src/librbd/cache/ParentCacheObjectDispatch.cc

```diff
@@ -80,27 +80,19 @@ bool ParentCacheObjectDispatch<I>::read(
   /* if RO daemon still don't startup, or RO daemon crash,
    * or session occur any error, try to re-connect daemon.*/
   if (!m_cache_client->is_session_work()) {
-    {
-      Mutex::Locker locker(m_lock);
-      if (m_connecting.load()) {
-        ldout(cct, 5) << "Parent cache is re-connecting RO daemon, "
-                      << "dispatch current request to lower object layer " << dendl;
-        return false;
+    if (!m_connecting.exchange(true)) {
+      if (!m_cache_client->is_session_work()) {
+        Context* on_finish = new FunctionContext([this](int ret) {
+          m_connecting.store(false);
+        });
+        create_cache_session(on_finish, true);
       }
```

Comment:
Need an ```else``` branch where you set ```m_connecting``` back to ```false``` if it's working

Reply:
i got your points, modified codes and added some comments to explain this race condition. thanks jason.